server/layout_test.go

```diff
@@ -1,79 +1,183 @@
 package server_test
 
-/*
-func TestNewLayout(t *testing.T) {
-	t.Parallel()
-	var tests = []struct {
-		Desc            string
-		AddError        error
-		ExistingLayouts map[string]chronograf.Layout
-		NewLayout       *models.Layout
-		ExpectedID      int
-		ExpectedHref    string
-		ExpectedStatus  int
+import (
+	"context"
+	"encoding/json"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/chronograf"
+	"github.com/influxdata/chronograf/mocks"
+	"github.com/influxdata/chronograf/server"
+)
+
+func Test_Layouts(t *testing.T) {
+	layoutTests := []struct {
+		name       string
+		expected   chronograf.Layout
+		allLayouts []chronograf.Layout
+		focusedApp string // should filter all layouts to this app only
+		shouldErr  bool
 	}{
 		{
-			Desc:     "Test that an error in datastore returns 500 status",
-			AddError: errors.New("error"),
-			NewLayout: &models.Layout{
-				Measurement: new(string),
-				App:         new(string),
-				Cells: []*models.Cell{
-					&models.Cell{
-						X: new(int32),
-						Y: new(int32),
-						W: new(int32),
-						H: new(int32),
-					},
+			"empty layout",
+			chronograf.Layout{},
+			[]chronograf.Layout{},
+			"",
+			false,
+		},
+		{
+			"several layouts",
+			chronograf.Layout{
+				ID:          "d20a21c8-69f1-4780-90fe-e69f5e4d138c",
+				Application: "influxdb",
+				Measurement: "influxdb",
+			},
+			[]chronograf.Layout{
+				chronograf.Layout{
+					ID:          "d20a21c8-69f1-4780-90fe-e69f5e4d138c",
+					Application: "influxdb",
+					Measurement: "influxdb",
 				},
 			},
-			ExpectedStatus: http.StatusInternalServerError,
+			"",
+			false,
 		},
 		{
-			Desc:            "Test that creating a layout returns 201 status",
-			ExistingLayouts: map[string]chronograf.Layout{},
-			NewLayout: &models.Layout{
-				Measurement: new(string),
-				App:         new(string),
-				Cells: []*models.Cell{
-					&models.Cell{
-						X: new(int32),
-						Y: new(int32),
-						W: new(int32),
-						H: new(int32),
+			"filtered app",
+			chronograf.Layout{
+				ID:          "d20a21c8-69f1-4780-90fe-e69f5e4d138c",
+				Application: "influxdb",
+				Measurement: "influxdb",
+			},
+			[]chronograf.Layout{
+				chronograf.Layout{
+					ID:          "d20a21c8-69f1-4780-90fe-e69f5e4d138c",
+					Application: "influxdb",
+					Measurement: "influxdb",
+				},
+				chronograf.Layout{
+					ID:          "b020101b-ea6b-4c8c-9f0e-db0ba501f4ef",
+					Application: "chronograf",
+					Measurement: "chronograf",
+				},
+			},
+			"influxdb",
+			false,
+		},
+		{
+			"axis zero values",
+			chronograf.Layout{
+				ID:          "d20a21c8-69f1-4780-90fe-e69f5e4d138c",
+				Application: "influxdb",
+				Measurement: "influxdb",
+				Cells: []chronograf.Cell{
+					{
+						X:    0,
+						Y:    0,
+						W:    4,
+						H:    4,
+						I:    "3b0e646b-2ca3-4df2-95a5-fd80915459dd",
+						Name: "A Graph",
+						Axes: map[string]chronograf.Axis{
```

Comment:
This part is the expected output, and this particular test is to ensure that the response looks like...

```json
{
    "axes": {
        "x": {
            "bounds": [],
            "label": ""
        },
        "y": {
            "bounds": [],
            "label": "",
        },
        "y2": {
            "bounds": [],
            "label": ""
        },
    }
```

...in the case where there are no axes set on an object. Setting `Label: ""` is identical to omitting it entirely in this case because of the zero value rules. I can add those in if you think it would be helpful for documentation purposes.